data/provider/src/test/java/tech/pegasys/teku/api/ChainDataProviderTest.java

```diff
@@ -552,7 +638,9 @@ public static Stream<Arguments> getRandaoIndexCases() {
     return args.stream();
   }
 
-  private ChainDataProvider setupAltairState() {
+  private Pair<
+          ChainDataProvider, tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState>
+      setupAltairState() {
```

Comment:
I think I'd stick with just returning the `ChainDataProvider` from here but add a method to get the internal state:

```java

  private tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState getHeadState() {
    return safeJoin(mockCombinedChainDataClient.getChainHead().orElseThrow().getState());
  }
```

It's still not really pretty but we generally avoid using generic `Pair` because it gets pretty confusing about what's actually in it given you only get generic `getLeft` and `getRight` type methods. And most places here only need the ChainDataProvider by itself.